components/image-builder-mk3/pkg/orchestrator/monitor.go

```diff
@@ -70,14 +71,17 @@ func (m *buildMonitor) Run() {
 	for {
 		wss, err := m.wsman.GetWorkspaces(ctx, &wsmanapi.GetWorkspacesRequest{
 			MustMatch: &wsmanapi.MetadataFilter{
-				Owner: buildWorkspaceOwnerID,
+				Annotations: map[string]string{
+					annotationManagedBy: buildWorkspaceManagerID,
```

Comment:
I have not tested the PR. Overall it looks good but I have a question about performance with this change. AFAIU filtering with owner is much faster and done on server side, but annotation based filters are done on client side.

If my understanding is correct, then how much this impacts the performance and in which cases

Reply:
Interesting, do you have a source for this? I haven't been able to find one.

Reply:
NVM this GetWorkspaces internally calls server component. I assumed that it was calling k8s api server directly.

BTW here is the [client go](https://github.com/kubernetes/client-go/blob/28ccde769fc5519dd84e5512ebf303ac86ef9d7c/listers/core/v1/pod.go#L50) implementation